Client/mods/deathmatch/logic/CResource.cpp

```diff
@@ -88,6 +88,10 @@ CResource::CResource(unsigned short usNetID, const char* szResourceName, CClient
 
 CResource::~CResource()
 {
+    // Force delete all objects in cache
+    // https://github.com/multitheftauto/mtasa-blue/issues/1840
+    g_pClientGame->GetElementDeleter()->DoDeleteAll();
+
```

Comment:
The entire purpose of having the element deleter is to call `CElementDeleter::DoDeleteAll` in pulses from `CClientGame::DoPulses`. There should be a better way to fix this.

Reply:
Then `stopResource` will crashes.
`CClientGame::DoPulses` contains `m_ElementDeleter.DoDeleteAll();` at 1135

Reply:
> Then stopResource will crashes.
Nope, if you do it properly. This code will fix one crash but introduce more crashes since you are forcefully deleting the objects. There's a reason why the entity deletion is delayed, i.e., to avoid dangling pointers in gta code by delaying deletion.

I investigated this crash. The `CBaseModelInfoInterface` instance is deleted before the object is internally deleted. You can use `std::shared_ptr<CClientModel>` to solve this. In `CClientObject::CClientObject`, you can increment the shared_ptr reference count by simply holding a shared_ptr and decrement shared_ptr reference count in `CClientObject::~CClientObject` by setting the shared_ptr varaible to nullptr.

In this way, when all objects are deleted, then the reference count of CClientModel will reach zero and the destructor will be called automatically.

Reply:
`~CClientObject` don't delete object from GTA:SA pool. So `~CClientObject` hide obejct from `CClientModel` and `CClientModel` can't replace this model with safe ID. MTA try delete object from pool, but can't, because `modelInfo` is invalid